### ISSUE TITLE
"is touching" reporting false negatives on tiny tests

### ISSUE THREAD

**Issue by cwillisf** [LLK/scratch-render#719]
### Expected Behavior

`is touching` tests should not depend on the size of the objects in question or the size of the stage

### Actual Behavior

This project behaves differently on staging vs production:
* Production: scratch.mit.edu/projects/445335326
* Staging: 1300006793

As of this test, the two environments differ only by these two changes:
* LLK/scratch-render#557
* LLK/scratch-render#624

### Steps to Reproduce

1. Load the project
2. Use WASD to move the cat onto & off of the purple dot
3. The cat should report "true" when overlapping the dot, and "false" when not

Actual behavior:
* in some cases on staging, the cat always reports "false"
* switching to full-screen stage and back may fix the issue
* the issue might depend on whether Scratch is running on a high-DPI screen or not

### Operating System and Browser

Seems to affect multiple browsers but Windows Chrome is the most-confirmed case so far.

**Comment by cwillisf** [comment]
After further testing, I think this is an existing issue that isn't specifically related to #557 or #624. In particular, I've tried this test project on many versions, going back into September, and I get "wrong" (always "false") results on all versions I've tested.

Here's the test project I've been using, from @BryceLTaylor (rename to SB3):
[Touching Color.zip](https://github.com/LLK/scratch-render/files/5491647/Touching.Color.zip)

**Comment by fsih** [comment]
I found that if I save the staging project to my computer, then re-upload it to scratch.mit.edu, it also always shows false. And vice versa, if I save the scratch.mit.edu project to my computer, and re-upload it to staging, it also shows true when I'm over the dot. I think the two projects are different

**Comment by fsih** [comment]
The not touching project on scratch.mit.edu:
![dot2](https://user-images.githubusercontent.com/2855464/98192704-81461880-1ee9-11eb-84c7-df479bee38d2.gif)
The touching project on scratch.ly:
![dot](https://user-images.githubusercontent.com/2855464/98192705-81deaf00-1ee9-11eb-9060-746792c0df78.gif)

**Comment by fsih** [comment]
The two purple dot costumes do differ
<img width="1087" alt="Screen Shot 2020-11-04 at 22 07 47" src="https://user-images.githubusercontent.com/2855464/98193215-a8e9b080-1eea-11eb-8c16-9b0864d4cef2.png">
Here's a project where you can switch between them and change the touching behavior (on either staging or prod)
https://scratch.mit.edu/projects/445446223/editor
I guess the slight translation makes the two rasterize differently

**Comment by fsih** [comment]
tangentially, what's up with those NaNs

**Comment by cwillisf** [comment]
Thanks for figuring this out!

Also: I feel like "what's up with those NaNs" is the topic about 90% of the time when NaN comes up in a conversation.

**Comment by BryceLTaylor** [comment]
I'm not sure how those costumes became different.  I just downloaded the project from staging and uploaded it to production without changing it.

**Comment by cwillisf** [comment]
Yeah - it seems possible that we have another bug in here related to costumes' bounds being slightly different on initial save vs. import & save again. Maybe the fix-up code that we run on import can cause small errors like this.
